spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceBucket.java

```diff
@@ -0,0 +1,163 @@
+import org.springframework.core.io.WritableResource;
+import org.springframework.util.Assert;
+
+/**
+ * Implements {@link WritableResource} for buckets in Google Cloud Storage.
+ */
+public class GoogleStorageResourceBucket implements WritableResource {
+
+	private final Storage storage;
+
+	private final String bucketName;
+
+	public GoogleStorageResourceBucket(Storage storage, String bucketName,
+			boolean createBucketIfNotExists) {
+		Assert.notNull(storage, "Storage object can not be null");
+		Assert.notNull(bucketName, "Bucket name can not be null");
+
+		this.storage = storage;
+		this.bucketName = bucketName;
+
+		if (!exists() && createBucketIfNotExists) {
+			this.storage.create(BucketInfo.newBuilder(this.bucketName).build());
+		}
+	}
+
+	@Override
+	public boolean exists() {
+		try {
+			return throwExceptionForNullBucket(resolve()).exists();
+		}
+		catch (IOException e) {
+			return false;
+		}
+	}
+
+	@Override
+	public boolean isReadable() {
+		return false;
+	}
+
+	@Override
+	public boolean isOpen() {
+		return false;
+	}
+
+	@Override
+	public URL getURL() throws IOException {
+		return getURI().toURL();
+	}
+
+	@Override
+	public URI getURI() throws IOException {
+		URI uri = null;
+		try {
+			uri = new URI(GoogleStorageProtocolResolver.PROTOCOL + getFilename());
+		}
+		catch (URISyntaxException e) {
+			throw new IOException("Invalid URI syntax", e);
+		}
+		return uri;
+	}
+
+	@Override
+	public File getFile() throws IOException {
+		throw new UnsupportedOperationException(
+				"Cannot get a file from a storage bucket.");
+	}
+
+	@Override
+	public long contentLength() throws IOException {
+		throw new UnsupportedOperationException(
+				"Cannot get content length for a storage bucket.");
+	}
+
+	@Override
+	public long lastModified() throws IOException {
+		throw new UnsupportedOperationException(
+				"Cannot get last-modified for a storage bucket.");
+	}
+
+	@Override
+	public Resource createRelative(String relativePath) throws IOException {
+		return new GoogleStorageResourceObject(this.storage,
+				getURI().toString() + (relativePath.startsWith("/") ? "" : "/")
+						+ relativePath);
+	}
+
+	@Override
+	public String getFilename() {
+		try {
+			return throwExceptionForNullBucket(resolve()).getName();
+		}
+		catch (IOException e) {
+			return null;
+		}
+	}
+
+	@Override
+	public String getDescription() {
+		return this.bucketName;
+	}
+
+	@Override
+	public InputStream getInputStream() throws IOException {
+		throw new UnsupportedOperationException(
+				"Cannot get input stream for a storage bucket.");
+	}
+
+	@Override
+	public boolean isWritable() {
+		return false;
+	}
+
+	@Override
+	public OutputStream getOutputStream() throws IOException {
+		throw new UnsupportedOperationException(
+				"Cannot get output stream for a storage bucket.");
+	}
+
+	private Bucket resolve() {
+		return this.storage.get(this.bucketName);
+	}
+
+	private Bucket throwExceptionForNullBucket(Bucket bucket) throws IOException {
```

Comment:
I know this code is copied and pasted, but should this method be called `getBucket()` instead?

Reply:
On a second look, it looks like this method is even unnecessary due to my previous comments.

Reply:
It is gone :)